src/network/MissionControlProtocol.cpp

```diff
@@ -403,137 +277,30 @@ MissionControlProtocol::MissionControlProtocol(SingleClientWSServer& server)
-}
-
-void MissionControlProtocol::updateArmIKRepeatTask() {
-	loguru::set_thread_name("MCP_ArmIK");
-	dataclock::time_point next_update_time = dataclock::now();
-	while (Globals::armIKEnabled) {
-		DataPoint<navtypes::Vectord<Constants::arm::IK_MOTORS.size()>> armJointPositions =
-			robot::getMotorPositionsRad(Constants::arm::IK_MOTORS);
-		if (armJointPositions.isValid()) {
-			navtypes::Vectord<Constants::arm::IK_MOTORS.size()> targetJointPositions =
-				Globals::planarArmController.getCommand(dataclock::now(), armJointPositions);
-			targetJointPositions /=
-				M_PI / 180.0 / 1000.0; // convert from radians to millidegrees
-			for (size_t i = 0; i < Constants::arm::IK_MOTORS.size(); i++) {
-				robot::setMotorPos(Constants::arm::IK_MOTORS[i],
-								   static_cast<int32_t>(targetJointPositions(i)));
-			}
-		}
-		next_update_time += Constants::ARM_IK_UPDATE_PERIOD;
-		std::this_thread::sleep_until(next_update_time);
+	if (enabled) {
+		_arm_ik_task.start();
+	} else {
+		_arm_ik_task.stop();
 	}
-}
 
-void MissionControlProtocol::telemReportTask() {
-	loguru::set_thread_name("MCP_Telem");
-	dataclock::time_point pt = dataclock::now();
-
-	while (true) {
-		for (const auto& cur : robot::types::name_to_jointid) {
-			robot::types::DataPoint<int32_t> jpos = robot::getJointPos(cur.second);
-			if (jpos.isValid()) {
-				auto jointNameFrznStr = cur.first;
-				std::string jointNameStdStr(jointNameFrznStr.data(), jointNameFrznStr.size());
-				sendJointPositionReport(jointNameStdStr, jpos.getData());
-			}
-		}
-
-		sendRoverPos();
-
-		pt += TELEM_REPORT_PERIOD;
-		std::this_thread::sleep_until(pt);
+	if (sendReport) {
+		sendArmIKEnabledReport(enabled);
 	}
 }
 
-void MissionControlProtocol::videoStreamTask() {
-	loguru::set_thread_name("MCP_Video");
-	while (this->_streaming_running) {
-		std::shared_lock<std::shared_mutex> stream_lock(this->_stream_mutex);
-		// for all open streams, check if there is a new frame
-		for (const auto& stream : _open_streams) {
-			const CameraID& cam = stream.first;
-			const uint32_t& frame_num = stream.second;
-			if (robot::hasNewCameraFrame(cam, frame_num)) {
-				// if there is a new frame, grab it
-				auto camDP = robot::readCamera(cam);
-
-				if (camDP) {
-					auto data = camDP.getData();
-					uint32_t& new_frame_num = data.second;
-					cv::Mat frame = data.first;
-					// update the previous frame number
-					this->_open_streams[cam] = new_frame_num;
-					const auto& encoder = this->_camera_encoders[cam];
-
-					// convert frame to encoded data and send it
-					auto data_vector = encoder->encode_frame(frame);
-					sendCameraStreamReport(cam, data_vector);
-				}
-			}
-
-			// break out of the loop if we should stop streaming
-			if (!this->_streaming_running) {
-				break;
-			}
-		}
-		std::this_thread::yield();
-	}
+void MissionControlProtocol::setRequestedJointPower(jointid_t joint, double power) {
+	_power_repeat_task.setJointPower(joint, power);
+	robot::setJointPower(joint, power);
```

Comment:
Doesn't the `periodicTask` of the `PowerRepeatTask` already call this method?

Reply:
Yeah, but it's delayed by up to one loop period. We want the joint power to be set immediately, to avoid any lag.